pennylane_lightning/core/src/observables/Observables.hpp

```diff
@@ -255,11 +286,29 @@ class HermitianObsBase : public Observable<StateVectorT> {
         [[maybe_unused]] StateVectorT &sv,
         [[maybe_unused]] std::vector<std::vector<PrecisionT>> &eigenValues,
         [[maybe_unused]] std::vector<size_t> &ob_wires) const override {
-        // TODO support. This support requires an additional method to solve
-        // eigenpair and unitary matrices, and the results of eigenpair and
-        // unitary matrices data need to be added to the Hermitian class and
-        // public methods are need to access eigen values.
-        PL_ABORT("Hermitian observables do not support shot measurement.");
+#ifdef PL_USE_LAPACK
+        std::vector<std::complex<PrecisionT>> mat(matrix_.size());
+
+        std::transform(matrix_.begin(), matrix_.end(), mat.begin(),
+                       [](ComplexT value) {
+                           return static_cast<std::complex<PrecisionT>>(value);
+                       });
+
+        PL_ABORT_IF_NOT(
+            Pennylane::Util::is_Hermitian<PrecisionT>(Util::exp2(wires_.size()),
+                                                      Util::exp2(wires_.size()),
+                                                      mat) == true,
+            "The matrix passed to HermitianObs is not a Hermitian matrix.");
+
+        eigenValues.clear();
+        ob_wires.clear();
+        ob_wires = wires_;
+        sv.applyMatrix(unitary_, wires_);
+        eigenValues.push_back(eigenVals_);
```

Comment:
consider using "shirnk_to_fit" to reduce the capacity to size after this line:
```suggestion
        eigenValues.push_back(eigenVals_);
        eigenValues.shirnk_to_fit();
```
https://en.cppreference.com/w/cpp/container/vector/shrink_to_fit

Reply:
Thanks! Since `eigenValues` is a `std::vector<std::vector<T>>` type, `std::vector<T>` doesn't meet the type requirement here. So I will keep current implementation.